### ISSUE TITLE
RFC: patchdir variable naming directory with patches

### ISSUE THREAD

**Issue by ackalker** [voidlinux/void-packages#5770]
I would like your opinion on introducing a new variable for use in templates, suggested name`patch_dirs`, default value: `patches patches-${XBPS_TARGET_MACHINE}` (values relative to the template directory).

The idea is for the patch step of the build to look in these directories for patches, thereby eliminating the need for (ugly, IMHO) code in templates to apply arch-specific patches.

Instead, templates could append to or override the default value as needed.

I would welcome any ideas and constructive criticism on this.

**Comment by Duncaen** [comment]
If your patch is architecture or libc specific the patch is wrong, I don't think we should make it more simple to apply wrong patches 😜.

**Comment by chneukirchen** [comment]
The problem is that when we have too many selective patches, its very easy to have combinations that break.

It is imo better to write generic patches, which can use libc or architecture flags when they are specific.

**Comment by ackalker** [comment]
> It is imo better to write generic patches, which can use libc or architecture flags when they are specific.

Agreed, but sadly not what I see in the various templates, at least not with regard to musl patches.
